test/dotnet-new3.UnitTests/SharedHomeDirectory.cs

```diff
@@ -0,0 +1,103 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.InteropServices;
+using Microsoft.NET.TestFramework.Assertions;
+using Microsoft.NET.TestFramework.Commands;
+using Xunit.Abstractions;
+using Xunit.Sdk;
+
+namespace dotnet_new3.UnitTests
+{
+    /// <summary>
+    /// This class represents shared /tmp/RANDOM-GUID/.templateengine/dotnetcli-preview/ folder
+    /// shared between multiple unit tests in same class, this is so each test
+    /// doesn't have to install everything from 0. To save some time executing tests.
+    /// </summary>
+    public class SharedHomeDirectory : IDisposable
+    {
+        private readonly ITestOutputHelper log;
+
+        public string HomeDirectory { get; } = Helpers.CreateTemporaryFolder("Home");
+        public string HomeVariable { get; } = Helpers.HomeEnvironmentVariableName;
+
+        public SharedHomeDirectory(IMessageSink messageSink)
+        {
+            this.log = new SharedTestOutputHelper(messageSink);
+            Initialize();
+        }
+
+        void Initialize()
+        {
+            var dn3Path = Environment.GetEnvironmentVariable("DN3");
+            if (string.IsNullOrEmpty(dn3Path))
+            {
+                var path = typeof(AllProjectsWork).Assembly.Location;
+                while (path != null && !File.Exists(Path.Combine(path, "Microsoft.TemplateEngine.sln")))
+                {
+                    path = Path.GetDirectoryName(path);
+                }
+                if (path == null)
+                    throw new Exception("Couldn't find repository root, because \"Microsoft.TemplateEngine.sln\" is not in any of parent directories.");
+                // DummyFolder Path just represents folder next to "artifacts" and "template_feed", so paths inside
+                // defaultinstall.package.list correctly finds packages in "../artifacts/"
+                // via %DN3%\..\template_feed\ or %DN3%\..\artifacts\packages\Microsoft.TemplateEngine.Core.*
+                path = Path.Combine(path, "DummyFolder");
+                dn3Path = path;
```

Comment:
shouldn't dn3path actually point to dll location?

Reply:
In this context %DN3% is supposed to represent folder path that is side-by-side with `template_feed` and `artifacts` folder, so `defaultinstall.package.list` entries that look like `%DN3%\..\artifacts\packages\Microsoft.TemplateEngine.Core.*` find packages and templates.